File: js/__tests__/josephus.test.js
// 约瑟夫问题算法测试
const josephus = require('../algorithm/josephus')

describe('约瑟夫环问题', () => {
    it('case1', () => {
        const safeArr = josephus()
        expect(safeArr).toEqual([1])
    })
    it('case2', () => {
        const safeArr = josephus({
            total: 41,
            step: 3,
            stayNum: 2
        })
        // 找到安全的两个位置
        expect(safeArr).toEqual([16, 31]);
    })
})

File: py/__tests__/josephus_test.py
import unittest
from algorithm.josephus import josephus


class TestJosephus(unittest.TestCase):
    def test_case_1(self):
        stay_arr = josephus()
        self.assertEqual(stay_arr, [1])

    def test_case_2(self):
        stay_arr = josephus(total=41, step=3, stay_num=2)
        self.assertEqual(stay_arr, [16, 31])
